CLAUDE.md

```diff
@@ -0,0 +1,92 @@
+# Claude Code Memory for Notecard Schema Repository
+
+## Repository Overview
+This repository contains JSON schemas for the Notecard API, following JSON Schema Draft 2020-12 specification. The schemas are used to generate the [Notecard API Reference Documentation](https://dev.blues.io/api-reference/).
+
+## Schema Creation Process
+When adding a new schema (e.g., `card.sleep`):
+
+1. **Create schema files**: Both `.req.notecard.api.json` and `.rsp.notecard.api.json` files
+2. **Add comprehensive tests**: Create corresponding test files in `tests/` directory
+3. **Update index**: Add schema references to `notecard.api.json`
+4. **Verify descriptions**: Ensure all descriptions match exactly with official API documentation
+
+## Repository Structure
+- Root directory contains all schema files (e.g., `card.*.json`)
+- `tests/` directory contains test files following pattern `test_card_*_req.py` and `test_card_*_rsp.py`
+- `notecard.api.json` serves as the main index file referencing all schemas
+
+## Testing
+- Uses `pipenv` for Python environment management
+- Tests use `pytest` with `jsonschema` validation
+- Run tests with: `pipenv run pytest`
+- All schemas must pass validation tests before being considered complete
+
+## Schema Conventions
+- **Title**: Follow pattern `"card.* Request/Response Application Programming Interface (API) Schema"`
+- **Version**: Currently using `"0.2.1"`
+- **API Version**: Currently `"9.1.1"`
+- **SKUs**: Include appropriate Notecard compatibility (e.g., `["WIFI"]` for WiFi-only APIs)
+- **Properties**: Support both `req` and `cmd` patterns using `oneOf` validation
+- **Additional Properties**: Set to `false` for strict validation
+
+## Custom Schema Fields
+The repository uses several custom fields for documentation generation:
+
+### `annotations`
+Used for notes, warnings, and deprecation notices:
+```json
+"annotations": [
+    {
+        "title": "note",
+        "description": "Only valid for Notecard WiFi v2. Sleep mode will not activate while USB-connected."
+    }
+]
+```
+
+### `samples`
+Provides JSON examples with titles and descriptions:
+```json
+"samples": [
+    {
+        "title": "Enable Sleep Mode",
+        "description": "Enable sleep mode with default settings.",
+        "json": "{\"req\": \"card.sleep\", \"on\": true}"
+    }
+]
+```
+
+### `sub-descriptions`
+Detailed descriptions for enum values:
```

Comment:
This is not always enum values, especially when multiple values can be supplied. Then it becomes a complex pattern.

```suggestion
Detailed descriptions for known parameter values:
```